File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0 , buy = prices[0];
        int n = prices.length ;
        for(int price : prices){
            if( buy > price ){
                buy = price;
            }else{
                profit = Math.max(profit , price-buy);
            }
        } 
        return profit ;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        profit = 0 
        sell =  prices[0]
        for i in prices :
            sell = i
            if i<buy :
                buy = i
                sell = i
            profit = max(profit,sell-buy)
        return profit
            